core/src/main/java/org/aerogear/mobile/core/http/OkHttpResponse.java

```diff
@@ -46,15 +51,31 @@ private synchronized void runCompletionHandler() {
         }
     }
 
+    private synchronized void runErrorHandler() {
+        if (errorHandler != null) {
+            errorHandler.run();
+            errorHandler = null;
+        }
+    }
+
     @Override
     public HttpResponse onComplete(Runnable completionHandler) {
         this.completionHandler = completionHandler;
-        if (response != null) {
+        if (response != null && response.isSuccessful()) {
```

Comment:
Ok, understand. Then you would see the completion handler more as soemthing like a `finally`? I'm ok with handling that in the completion handler but then we should probably remove the error handler because you would have to handle a possible error in your completion handler anyway.